onadata/libs/authentication.py

```diff
@@ -113,7 +114,8 @@ def authenticate_credentials(self, key):
         try:
             if isinstance(key, bytes):
                 key = key.decode('utf-8')
-            token = self.model.objects.get(key=key)
+            with use_master:
+                token = self.model.objects.get(key=key)
```

Comment:
This means that we will always hit the master database for this request. It would be ideal to only read the master database only in the situation where the file does not exist in the read replica.

```python
try:
    token = self.model.objects.get(key=key)
except self.model.DoesNotExist:
    invalid_token = True
    if getattr(settings, 'SLAVE_DATABASES', []):
        try:
            with use_master:
                token = self.model.objects.get(key=key)
        except self.model.DoesNotExist:
            invalid_token = True
        else:
            invalid_token = False
    if invalid_token:
        raise exceptions.AuthenticationFailed(_(u'Invalid token'))
```

Reply:
Cool, this makes sense and looks better. I'll implement this